pkg/misc/utils.go

```diff
@@ -393,3 +400,15 @@ func GetRabbitMQCredentialsFor(cr *apiv1alpha1.Astarte, c client.Client) (string
 
 	return host, port, string(secret.Data[usernameKey]), string(secret.Data[passwordKey]), nil
 }
+
+// GetCassandraUserCredentialsSecret gets the secret holding Cassandra credentials in the form <secret name>, <username key>, <password key>
+func GetCassandraUserCredentialsSecret(cr *apiv1alpha1.Astarte) (string, string, string) {
+	if cr.Spec.Cassandra.Connection != nil {
+		if cr.Spec.Cassandra.Connection.Secret != nil {
+			return cr.Spec.Cassandra.Connection.Secret.Name, cr.Spec.Cassandra.Connection.Secret.UsernameKey, cr.Spec.Cassandra.Connection.Secret.PasswordKey
+		}
+	}
+
+	// Standard setup
+	return cr.Name + "-cassandra-user-credentials", CassandraDefaultUserCredentialsUsernameKey, CassandraDefaultUserCredentialsPasswordKey
```

Comment:
This should probably return empty strings or some bool to determine whether user has set up authentication or not.